ledger-api/grpc-definitions/com/daml/ledger/api/v1/version_service.proto

```diff
@@ -26,6 +28,74 @@ message GetLedgerApiVersionRequest {
 
 message GetLedgerApiVersionResponse {
 
-  // The version of the ledger API
+  // The version of the ledger API.
   string version = 1;
+
+  // The features supported by this Ledger API endpoint.
+  //
+  // Daml applications CAN use the feature descriptor on top of 
+  // version constraints on the Ledger API version to determine
+  // whether a given Ledger API endpoint supports the features
+  // required to run the application.
+  //
+  // See the feature descriptions themselves for the relation between
+  // Ledger API versions and feature presence.
+  FeatureDescriptor features = 2;
+}
+
+message FeatureDescriptor {
+  // Features that are determined by the ledger backing the participant node
+  // that serves the Ledger API endpoint.
+  //
+  // All participant nodes connected to the same ledger are expected to report the
+  // same values for the ledger features.
+  LedgerFeatures ledger = 1;
+
+  // Features that are are specific to the participant node that serves
+  // the Ledger API endpoint. 
+  ParticipantFeatures participant = 2;
+
+  // Features under development or features that are used
+  // for ledger implementation testing purposes only.
+  //
+  // Daml applications SHOULD not depend on these in production.
+  ExperimentalFeatures experimental = 3;
+}
+
+// See the feature message definitions for descriptions.
+message LedgerFeatures {
+  FeatureOrderedContractIds ordered_contract_ids = 2;
 }
+
+// See the feature message definitions for descriptions.
+message ParticipantFeatures {
+  FeatureDamlLf daml_lf = 1;
+  FeatureParticipantPruning participant_pruning = 2;
+}
```

Comment:
Add the FeatureCausalOrdering for whether the participant outputs transactions in causal order.

Reply:
So KV ledgers output transactions in causal order, while Canton can have use-after-archive cases can't it? However, it can't have 'use-before-create' and 'archive-before-create'. So it's not that there is no causal order, but a weaker causal order isn't it?

I'm wondering whether there's a better name like `FeatureParticipantStrongTransactionOrder` that does not make a client equally nervous if it is not present.

Reply:
Canton in single-domain mode always achieves causal ordering. In a multi-domain setting, Canton currently doesn't enforce causality, so you can observe both archive-before-create and use-before-create. @phoebe-da is about to add causality enforcement to Canton, but for the time being, we consider this a configuration option because enforcing causality may create liveness problems.